### PR TITLE
WIP - sync between payment and opportunity allocations

### PR DESCRIPTION
**Pull request by njjc**
# Critical Changes

# Changes

- We added syncing between Payment and Opportunity Allocations to support a future enhancement. The feature is disabled by default and should have no visible impact.

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: src/classes/ALLO_PaymentSync_TEST.cls [hunk 1]
```diff
@@ -202,7 +202,7 @@ private with sharing  class ALLO_PaymentSync_TEST {
         insert alloForInsert;
 
         List<Allocation__c> queryAllocations = getAllocationsOrderByAmount();
-        System.assertEquals(6, queryAllocations.size(), 'Six allocations should exist.');
+        System.assertEquals(9, queryAllocations.size(), 'Nine allocations should exist.');
```
Comment (jbrandolinisf) [inline src/classes/ALLO_PaymentSync_TEST.cls]: I made the same change to my branch.

**Comment by annejackson** [comment]
Per @ameyersalesforce, we need to remove the Payment Allocations setting UI (and it's related text, etc.) from the NPSP Settings->Donation page. (We will need to make sure that once this PR is merged- feature/payment-allocations__pmt-to-opp-sync it doesn't have it-also will it need to be removed from feature/payment-allocations branch @jbrandolinisf ?)

**Comment by annejackson** [comment]
@melissabarber and @markross-sfdo- In the "Changes" section of this PR, I put what we had for the original Payment Allocations changes/release notes and added a few words about syncing. Feel free to add/edit/remove to your liking!

**Comment by jbrandolinisf** [comment]
Superceded by PR: https://github.com/SalesforceFoundation/Cumulus/pull/4211

**Pull request closed without merging**